pkg/apis/pipeline/v1beta1/matrix_types.go

```diff
@@ -51,51 +57,62 @@ type MatrixInclude struct {
 	Params Params `json:"params,omitempty"`
 }
 
+type Combination map[string]string
+type Combinations []Combination
+
 // FanOut produces combinations of Parameters of type String from a slice of Parameters of type Array.
-func (m *Matrix) FanOut() []Params {
-	var combinations []Params
+func (m *Matrix) FanOut() {
+	var combinations Combinations
 	for _, parameter := range m.Params {
-		combinations = fanOut(parameter, combinations)
+		combinations = combinations.fanOut(parameter)
 	}
-	return combinations
+	m.Combinations = combinations
+}
+
+func (c Combinations) FormatCombinations() []Params {
```

Comment:
```suggestion
func (c Combinations) Format() []Params {
```

to remove repetition

Reply:
sure, changed it to `ToParams`, doesn't avoid repetition though 🤣